transformer/src/main/scala/com/snowplowanalytics/snowflake/transformer/TransformerJob.scala

```diff
@@ -52,13 +52,33 @@ object TransformerJob {
     sc.register(keysAggregator)
 
     val events = sc
-      .textFile(jobConfig.input)
-      .map { e => Transformer.jsonify(e) }
+      .wholeTextFiles(jobConfig.input)
+      .flatMap(rdd => {
+        val fileName = rdd._1
+        val fileContents = rdd._2
+        fileContents.lines.zipWithIndex.map{case (event, index) =>
+          val lineNumber = index + 1
+          try {
+            Right(Transformer.jsonify(event))
+          } catch {
+            case ex : Throwable =>
+              log.error(s"Could not parse event on line $lineNumber in file $fileName")
+              log.error(s"${ex.getMessage}")
+              log.error(s"$event")
```

Comment:
Might be better to merge all the 3 logs here. If the code runs concurrently, might be hard _e.g._ to know which exception message relates to which event.

Reply:
Good catch, changed this to a single log statement of a multiline string.